src/chapters/aspdotnet-model-validation/validation-attributes.rst

```diff
@@ -0,0 +1,184 @@
+      public string Password { get; set; }
+
+We will see how to ensure these error messages are properly displayed in the next section, :ref:`validating-models`.
+
+Applying Validation Attributes - Video
+--------------------------------------
+
+.. TODO: Add video here on what attributes to add 
+
+.. admonition:: Note
+
+   If you want to verify what code this video starts with, check out the `adding-viewmodels <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/adding-viewmodels>`__ branch.
+   If you want to verify what code this video ends with, check out the `validation-attributes <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/validation-attributes>`__ branch.
+
+Applying Validation Attributes - Text
+-------------------------------------
+
+To configure validation on the model-side, we begin by adding validation attributes to each field to which we want to apply constraints.
+
+For our ``AddEventViewModel`` class, we add ``[StringLength]`` and ``[Required]`` to the ``Name`` property, and just ``[StringLength]`` to the ``Description`` property.
+
+.. sourcecode:: csharp
+   :lineno-start: 8
+
+   [Required(ErrorMessage = "Name is required.")]
+   [StringLength(50, MinimumLength = 3, ErrorMessage = "Name must be between 3 and 50 characters")]
+   public string Name { get; set; }
+
+   [StringLength(500, ErrorMessage = "Description too long!")]
+   public string Description { get; set; }
+
+The required ``MaximumLength`` and optional ``MinimumLength`` parameters for ``[StringLength]`` specify the maximum and minimum number of allowed characters, respectively.
+Omitting either of these means that no min or max will be applied for the field.
+For our ``Description`` field, leaving off ``[Required]`` makes this field optional.
+
+Each of our attributes also receives a ``ErrorMessage`` parameter, which provides a user-friendly message to display to the user if the particular validation rule fails.
+We will see how to display these in a view a bit later. 
+
+Next, we add a new property to store a contact email for each event.
+This is a ``string`` named ``ContactEmail``.
+Validating email addresses by directly applying each of the rules that an email must satisfy is *extremely* difficult.
+Thankfully, there is an ``[EmailAddress]`` validation attribute that we can apply to our new field.
+
+.. sourcecode:: csharp
+   :lineno-start: 8
+
+      [Required(ErrorMessage = "Name is required.")]
+      [StringLength(50, MinimumLength = 3, ErrorMessage = "Name must be between 3 and 50 characters")]
+      public string Name { get; set; }
+
+      [StringLength(500, ErrorMessage = "Description too long!")]
+      public string Description { get; set; }
+
+      [EmailAddress(ErrorMessage = "Invalid email. Try again.")]
+      public string ContactEmail { get; set; }
+
+Before we can start up our application, we need to add a new input to our form to take in the contact email for an event organizer.
```

Comment:
The code for the new email input isn't shown here. Even if it's very straightforward code, I suggest including it since we are generally showing all new code segments in sections like this.

Reply:
Done!